src/tiler/server.js

```diff
@@ -103,9 +103,11 @@ var config = {
             }
 
             if (body) {
-                s3Obj
-                    .upload({Body: body})
-                    .send(function(err, data) { throw (err); });
```

Comment:
Perhaps the original code was executing differently than the example?

```javascript
s3obj.upload({Body: body}).
  on('httpUploadProgress', function(evt) { console.log(evt); }).
  send(function(err, data) { console.log(err, data) });
```

Looks like that has `on()` chained with `send()`, but the original didn't. Not super familiar with how this chaining behaves, but potentially `send()` was being called in cases where there was no data?